llvm/lib/Target/X86/X86InstrSystem.td

```diff
@@ -68,6 +68,12 @@ def SYSENTER : I<0x34, RawFrm, (outs), (ins), "sysenter", []>, TB;
 def SYSEXIT   : I<0x35, RawFrm, (outs), (ins), "sysexit{l}", []>, TB;
 def SYSEXIT64 :RI<0x35, RawFrm, (outs), (ins), "sysexitq", []>, TB,
                   Requires<[In64BitMode]>;
+
+// FRED Instructions
+def ERETS: I<0x01, MRM_CA, (outs), (ins), "erets",
```

Comment:
> The following sections indicate situations in which one of the instructions may cause a
> control-protection exception (#CP). 

We should set hasSideEffects = 1 for them

Reply:
[0c6410e](https://github.com/llvm/llvm-project/pull/91909/commits/0c6410e012eea7629a91796d32aaf5bd35391356)